dev/devicelab/lib/framework/cocoon.dart

```diff
@@ -73,15 +100,59 @@ class Cocoon {
       print('${rec.level.name}: ${rec.time}: ${rec.message}');
     });
 
-    final Map<String, dynamic> status = <String, dynamic>{
+    final Map<String, dynamic> updateRequest = _constructUpdateRequest(
+      gitBranch: gitBranch,
+      builderName: builderName,
+      result: result,
+    );
+    final Map<String, dynamic> response = await _sendCocoonRequest('update-task-status', updateRequest);
+    if (response['Name'] != null) {
+      logger.info('Updated Cocoon with results from this task');
+    } else {
+      logger.info(response);
+      logger.severe('Failed to updated Cocoon with results from this task');
+    }
+  }
+
+  /// Write the given parameters into an update task request and store the JSON in [resultsPath].
+  Future<void> writeTaskResultToFile({
+    @required String builderName,
+    @required String gitBranch,
+    @required TaskResult result,
+    @required String resultsPath,
+  }) async {
+    assert(builderName != null);
+    assert(gitBranch != null);
+    assert(result != null);
+    assert(resultsPath != null);
+
+    final Map<String, dynamic> updateRequest = _constructUpdateRequest(
+      gitBranch: gitBranch,
+      builderName: builderName,
+      result: result,
+    );
+    final File resultFile = fs.file(resultsPath);
+    if (resultFile.existsSync()) {
```

Comment:
shall we try removing and creating again? Otherwise, we will lose metrics from this task run.

Reply:
Done. It'll delete and recreate the file.